Source/missiles.cpp

```diff
@@ -547,40 +547,21 @@ void CheckMissileCol(Missile &missile, int mindam, int maxdam, bool shift, Point
 			}
 		}
 	}
-	if (dObject[mx][my] != 0) {
-		Object &object = Objects[abs(dObject[mx][my]) - 1];
-		if (!object._oMissFlag) {
-			if (object.IsBreakable()) {
-				BreakObject(-1, object);
-			}
-			if (!nodel)
-				missile._mirange = 0;
-			missile._miHitFlag = false;
+	if (IsMissileBlockedByTile({ mx, my })) {
+		Object *object = ObjectAtPosition({ mx, my });
+		if (object != nullptr && object->IsBreakable()) {
+			BreakObject(-1, *object);
 		}
-	}
-	if (nMissileTable[dPiece[mx][my]]) {
+
 		if (!nodel)
 			missile._mirange = 0;
 		missile._miHitFlag = false;
 	}

```

Comment:
This isn't strictly equivalent to the old code, but I don't think there are any naturally occurring cases where a breakable object that allows missiles to pass through is on a tile which blocks missiles... that'd have to be something like a barrel inside a wall or set piece.